src/devices/Tlc1543/Tlc1543.cs

```diff
@@ -0,0 +1,289 @@
+        /// <param name="DOUT">Data Out</param>
+        /// <param name="EOC">End of Conversion</param>
+        /// <param name="IOCLK">I/O Clock</param>
+        public Tlc1543(int ADDR, int CS, int DOUT, int IOCLK, int EOC)
+        {
+            _address = ADDR;
+            _chipSelect = CS;
+            _dataOut = DOUT;
+            _endOfConversion = EOC;
+            _inputOutputClock = IOCLK;
+
+            _digital.OpenPin(_address, PinMode.Output);
+            _digital.OpenPin(_chipSelect, PinMode.Output);
+            _digital.OpenPin(_dataOut, PinMode.InputPullUp);
+            _digital.OpenPin(_endOfConversion, PinMode.InputPullUp);
+            _digital.OpenPin(_inputOutputClock, PinMode.Output);
+        }
+
+        #endregion
+
+        #region Methods
+
+        /// <summary>
+        /// Reads sensor value
+        /// </summary>
+        /// <param name="channelNumber">Channel to be read</param>
+        /// <returns>A 10 bit value corresponding to relative voltage level on specified device channel</returns>
+        public int ReadChannel(Channel channelNumber)
+        {
+            Read((int)channelNumber);
+            return Read((int)ChargeChannel);
+        }
+
+        /// <summary>
+        /// Reads sensor values into an List
+        /// </summary>
+        /// <param name="channelList">List of channels to read</param>
+        /// <returns>List of 10 bit values corresponding to relative voltage level on specified device channels</returns>
+        public List<int> ReadChannels(List<Channel> channelList)
+        {
+            List<int> values = new List<int>(channelList.Count);
+            Read((int)channelList[0]);
+            for (int i = 1; i < channelList.Count; i++)
+            {
+                values.Add(Read((int)channelList[i]));
+            }
+
+            values.Add(Read((int)ChargeChannel));
+            return values;
+        }
+
+        /// <summary>
+        /// Reads all sensor values into an List
+        /// </summary>
+        /// <returns>List of 10 bit values corresponding to relative voltage level on all channels</returns>
+        public List<int> ReadAll()
+        {
+            List<int> values = new List<int>(_channels);
+            Read(0);
+            for (int i = 1; i < values.Count; i++)
+            {
+                values.Add(Read(i));
+            }
+
+            values.Add(Read((int)ChargeChannel));
+            return values;
+        }
+
+        /// <summary>
+        /// Mode 1 (Fast mode):
+        /// <br>10 clock transfer with !CS high </br>
+        /// <br>Figure 9 in datasheet</br>
+        /// </summary>
+        /// <param name="channelNumber">Channel to read</param>
+        /// <returns>10 bit value corresponding to relative voltage level on channel</returns>
+        public int Read(int channelNumber)
+        {
+            int value = 0;
+            _digital.Write(_chipSelect, 0);
+            DelayHelper.DelayMicroseconds(1, false);

```

Comment:
Please refer to the documentation for the delays. see as well the recommendation for this topic: https://github.com/dotnet/iot/tree/master/src/devices#commenting-the-code